src/services/catalog.service.ts

```diff
@@ -152,6 +168,20 @@ export class CatalogService {
 
     return retCategories;
   }
+
+  private returnCatalogItems(deferred: any, catalogItems: any, numPromisesExecuted: number, totalNumPromises: number, errorMsg: string) {
+    if (numPromisesExecuted < totalNumPromises) {
+      return;
+    }
+
+    errorMsg = _.size(errorMsg) ? 'Error retrieving all Catalog Items. ' + errorMsg : null;
```

Comment:
This is an end user facing message, don't leak information about our underlying variable names.  Just say something like "Unable to load all content for the catalog." or "An error occurred loading the catalog contents."